src/nupic/frameworks/opf/common_models/cluster_params.py

```diff
@@ -29,14 +29,17 @@
 def getScalarMetricWithTimeOfDayAnomalyParams(metricData,
                                               minVal=None,
                                               maxVal=None,
-                                              minResolution=None):
+                                              minResolution=None,
+                                              tmImplementation = "cpp"):
   """
     Return a dict that can be used to create an anomaly model via OPF's
     ModelFactory.
 
     :param metricData: numpy array of metric data. Used to calculate minVal
       and maxVal if either is unspecified
 
+    :param tmImplementation: string specifying type of temporal memory implementation.
+
```

Comment:
Please specify the possible choices, otherwise the user won't know what to pass in.

This comment should also appear at the end, to match the order of arguments.